### ISSUE TITLE
avoid "Unused use statement" warnings for `use CRM_MyExtension_ExtensionUtil as E;`

### ISSUE THREAD

**Issue by twomice** [civicrm/civicrm-buildkit#535]
- Civix very handily adds the line `use CRM_MyExtension_ExtensionUtil as E;` at the top of most of its auto-generated files.
- Then civilint comes along and says "Unused use statement" if that file doesn't actually containe `E::ts()` or some other use of `E`.

I propose we do away with this warning in the case of these auto-generated lines, by avoiding the warning when the use is `as E`, like so

```
--- a/coder_sniffer/Drupal/Sniffs/Classes/UnusedUseStatementSniff.php
+++ b/coder_sniffer/Drupal/Sniffs/Classes/UnusedUseStatementSniff.php
@@ -71,6 +71,10 @@ class UnusedUseStatementSniff implements Sniff
             return;
         }
 
+        if ($tokens[$classPtr]['content'] == 'E') {
+            return;
+        }
+
         // Search where the class name is used. PHP treats class names case
         // insensitive, that's why we cannot search for the exact class name string
         // and need to iterate over all T_STRING tokens in the file.
```

**Comment by totten** [comment]
@twomice Oooh, I like that patch. I completely agree with this.

FWIW, there was [a recent patch in civix](https://github.com/totten/civix/commit/f13c141b6a1eb5ebd51cf554630ea3af16444881) that tried to work-around this, but it requires new messiness in the boilerplate, and it was only applied to one template. (There are 13 more templates referencing `E`, and then there's the existing codebase generated from the templates.) But I was too lazy to figure out to quiet the phpcs rule.

So this seems like a good idea. Ported: https://github.com/civicrm/coder/pull/11

**Comment by twomice** [comment]
Thanks @totten!

**Comment by MikeyMJCO** [comment]
Looks like this has been fixed with #546?

**Comment by seamuslee001** [comment]
I agree with Mikey going to close this out
